Stripe/STPPaymentContext.m

```diff
@@ -673,10 +667,17 @@ - (void)requestPayment {
                              onFinish:^(STPPaymentStatus status, NSError * _Nullable error) {
                                  [self.hostViewController dismissViewControllerAnimated:[self transitionAnimationsEnabled]
                                                                              completion:^{
-                                     [self didFinishWithStatus:status
-                                                         error:error];
+                                                                                 if (status == STPPaymentStatusUserCancellation &&
+                                                                                     self.paymentResult != nil) {
```

Comment:
I think you've got one or more race conditions in here. Here's my understanding of the happy path:

1. The user interacts with the `PKPaymentAuthorizationController`, authorizing the payment
2. stripe-ios creates a Source
3. stripe-ios attaches the Source to a Customer
4. The merchant's app receives `paymentContext:didCreatePaymentResult:completion:` and sends that payment result to their backend.
5. Merchant app calls `completion()` and then stripe-ios does some cleanup before calling `paymentContext:didFinishWithStatus:error:`

As far as I can tell, this PR handles the case where the user hits cancel during the processing of step 4.

However, I'm worried about what happens if the user hits cancel **earlier**. As far as I can tell, there's *nothing* that interrupts the subsequent steps. They're all chained together, and things are going to go off the rails.

As a result, the `STPPaymentContextDelegate` is going to get a very confusing set of callbacks. It'll be told that the context has finished with status canceled, and *then* it'll be told about the payment result, and presumably go send it to their backend.

Reply:
Also, the `paymentResult` will stick around on the `STPPaymentContext`, and if they re-use the context to try again it's going to throw off your new logic.